src/main/java/org/broadinstitute/hellbender/tools/walkers/conversion/GtfToBed.java

```diff
@@ -0,0 +1,303 @@
+
+        // put the interval, type as gene, and the name of gene
+        final GtfInfo gtfInfo = new GtfInfo(interval, GtfInfo.Type.GENE, gtfFeature.getGeneName());
+
+        // store in hashmap with key as geneId
+        featureInfoMap.put(gtfFeature.getGeneId(), gtfInfo);
+    }
+
+    // stores the transcript ID and Interval info in hashmap
+    private void processTranscriptFeature(GencodeGtfFeature gtfFeature) {
+        //get interval and put the interval, type as transcript, and the name of the gene it's in
+        final Interval interval = new Interval(gtfFeature.getContig(), gtfFeature.getStart(), gtfFeature.getEnd());
+        final GtfInfo gtfInfo = new GtfInfo(interval, GtfInfo.Type.TRANSCRIPT, gtfFeature.getGeneName());
+
+        //store in hashmap with key as transcriptId
+        featureInfoMap.put(gtfFeature.getTranscriptId(), gtfInfo);
+
+        //update start/end of corresponding gene if needed
+        updateGeneStart(gtfFeature);
+        updateGeneEnd(gtfFeature);
+    }
+
+    // update the gene interval start position based on the transcript
+    private void updateGeneStart(GencodeGtfFeature gtfFeature) {
+        // get the start value of the gene
+        int geneStart = featureInfoMap.get(gtfFeature.getGeneId()).getStart();
+
+        // if the transcript start is less than the gene start
+        if (gtfFeature.getStart() < geneStart) {
+            // set the gene start to be the transcript start
+            geneStart = gtfFeature.getStart();
+            updateGeneInterval(gtfFeature, geneStart, featureInfoMap.get(gtfFeature.getGeneId()).getEnd());
+        }
+    }
+
+    // update the gene interval end position based on the transcript
+    private void updateGeneEnd(GencodeGtfFeature gtfFeature) {
+        // get the end value of the gene
+        int geneEnd = featureInfoMap.get(gtfFeature.getGeneId()).getEnd();
+
+        // if the transcript start is greater than the gene start
+        if (gtfFeature.getEnd() > geneEnd) {
+            // set the gene end to be the transcript end
+            geneEnd = gtfFeature.getEnd();
+            updateGeneInterval(gtfFeature, featureInfoMap.get(gtfFeature.getGeneId()).getStart(), geneEnd);
+        }
+    }
+
+    // updates an interval of the gene if it needs to be changed
+    private void updateGeneInterval(GencodeGtfFeature gtfFeature, int geneStart, int geneEnd) {
+        Interval geneInterval = new Interval(gtfFeature.getContig(), geneStart, geneEnd);
+        GtfInfo gtfGeneInfo = new GtfInfo(geneInterval, GtfInfo.Type.GENE, gtfFeature.getGeneName());
+        featureInfoMap.put(gtfFeature.getGeneId(), gtfGeneInfo);
+    }
+
+    // runs immediately after it has gone through each line of gtf (apply method)
+    @Override
+    public Object onTraversalSuccess() {
+        // get dictionary
+        SAMSequenceDictionary sequenceDictionary = getBestAvailableSequenceDictionary();
```

Comment:
This could return null and maybe crash below when you try to get the sorted map. If there is no sequence dictionary then there is no way to really sort the outputs and thus its required. Currently both it and the alternative way of specifying it (using -R) are optional so a user could potentially hit a null pointer exception here if they forget to specify a dictionary source after the tool has already done most of the work processing their gtf.

Add a `customCommandLineValidation()` override or some code to `onTraversalStart()` that asserts that the sequence dictionary does in fact exist and if it doesnt returns a helpful `UserException` explaining what the user needs. 

In addition to this you should add two tests: the first one that specifies no sequence dictionary and and has `expectedExceptions = UserException.class` in the test block that asserts its failing gracefully. The second that uses -R on a reference to specify a sequence dictionary. I would also consider adding -R specification to one of the examples in the javadocs so its clear to users.